transmogrifier/sources/dspace_mets.py

```diff
@@ -0,0 +1,268 @@
+        for alternate_title in [t for t in all_titles if "type" in t.attrs]:
+            if alternate_title.string:
+                kwargs.setdefault("alternate_titles", []).append(
+                    timdex.AlternateTitle(
+                        value=alternate_title.string,
+                        kind=alternate_title["type"],
+                    )
+                )
+
+        # call_numbers: relevant field in DSpace (dc.subject.classification) is not
+        # mapped to the OAI-PMH METS output.
+
+        # citation
+        citation = xml.find("mods:identifier", type="citation")
+        kwargs["citation"] = citation.string if citation and citation.string else None
+
+        # content_type
+        content_types = xml.find_all("mods:genre")
+        kwargs["content_type"] = [
+            content_type.string for content_type in content_types if content_type.string
+        ] or None
+
+        # contents: relevant field in DSpace (dc.description.tableofcontents) is not
+        # mapped to the OAI-PMH METS output.
+
+        # contributors
+        contributors = xml.find_all("mods:name")
+        for contributor in contributors:
+            name = contributor.find("mods:namePart")
+            if name and name.string:
+                if role_field := contributor.find("mods:role"):
+                    role_name = role_field.find("mods:roleTerm")
+                    kind = role_name.string or "Contributor role not specified"
+                else:
+                    kind = "Contributor role not specified"
+                kwargs.setdefault("contributors", []).append(
+                    timdex.Contributor(
+                        kind=kind,
+                        value=name.string,
+                    )
+                )
+
+        # dates
+        # Only publication date is mapped from DSpace, other relevant date field (dc.
+        # coverage.temporal) is not mapped to the OAI-PMH METS output.
+        publication_date = xml.find("mods:dateIssued")
+        if publication_date and publication_date.string:
+            kwargs["dates"] = [
+                timdex.Date(kind="Publication date", value=publication_date.string)
+            ]
+
+        # edition field not used in DSpace
+
+        # file_formats
+        # Only maps formats with attribute use="ORIGINAL" because other formats such as
+        # USE="TEXT" are used internally by DSpace and not made publicly available.
+        file_groups = xml.find_all("fileGrp", USE="ORIGINAL")
+        for file_group in file_groups:
+            file = file_group.find("file")
+            if file and file.get("MIMETYPE"):
+                kwargs.setdefault("file_formats", []).append(file["MIMETYPE"])
+
+        # format
+        kwargs["format"] = "electronic resource"
+
+        # funding_information: relevant field in DSpace (dc.description.sponsorship) is
+        # not mapped to the OAI-PMH METS output.
+
+        # holdings field not used in DSpace
```

Comment:
There’s a lot of comments here and I understand some when clarification is needed (why a field may exist in DC but not the METS export) but some like this feel like what Clean Code was trying to dissuade us from creating and won’t likely be useful to future developers. If we want to retain this info, I think there are better methods of documentation than interspersed with the code. Non-blocking but I’d recommend a second look at whether all these are necessary

Reply:
Hmm. I think this is exactly what Clean Code was promoting: comments aren't for explaining _what_ the code does, but rather _why_. We don't expect those initial mapping spreadsheets to stick around, so documenting why some fields are mapped and some aren't is useful to have here. Knowing why `holdings` isn't mapped is just as useful as knowing why `funding_information` isn't mapped, regardless of the reason. And I'm not sure where else would be a better place for it, as going forward this codebase is the official source of truth for all of these data mappings.